platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/Collectors.java

```diff
@@ -201,6 +213,11 @@ public int hashCode() {
         public int size() {
             return Iterables.size(value);
         }
+
+        @Override
+        public String toString() {
+            return value.toString();
```

Comment:
❓ Shouldn't this be `Iterables.toString(value)`?

Reply:
Side-note: I was felling uncomfortable with the idea of iterating/consuming the iterable in `toString` then I saw it's already being done for `size` above.

Reply:
>Shouldn't this be Iterables.toString(value)?

Can be, though see below. I suppose, in practice most Iterables in the wild have sensible toString implementations. I wonder if we need to impose the limit on the number of items stringified.

> Side-note: I was felling uncomfortable with the idea of iterating/consuming the iterable in `toString` then I saw it's already being done for `size` above.

I see two corner cases, where it matters. A one-shot Iterable is a pretty rare thing, and an endless Iterable is going to be a problem anyway. `toString` is mostly for debugging, so exhausting a one-shot iterable it probably fine.

Interestingly, in both cases, not using `Iterables.toString` gives us a benefit of allowing the custom implementation to kick in and handle the problem in a meaningful way.